modules/vocabularies/src/main/frontend/src/vocabularyAction.jsx

```
@@ -74,6 +80,69 @@ const useStyles = makeStyles(theme => ({
 
 export default function VocabularyAction(props) {
   const classes = useStyles();
+  const [displayPopup, setDisplayPopup] = React.useState(false);
+  const [linkedQuestions, setLinkedQuestions] = useState([]);
+  const [questionnaires, setQuestionnaires] = useState([]);
+  const handleOpen = () => {fetchQuestionnaires();}
+  const handleClose = () => {setDisplayPopup(false);}
+  const handleUninstall = () => {setDisplayPopup(false); props.uninstall();}
+
+  let fetchQuestionnaires = () => {
+    if (questionnaires.length === 0) {
+      // Send a fetch request to determine the questionnaires available
+      const query = `select n.* from [lfs:Questionnaire] as n inner join [lfs:Question] as q on isdescendantnode(q, n) where q.sourceVocabulary='${props.acronym}'`;
+      fetch(`/query?query=${encodeURIComponent(query)}&limit=100`)
+        .then((response) => response.ok ? response.json() : Promise.reject(response))
```

A problem with fetches is that in case the user is no longer authenticated, the fetch will return the login page, and this will silently fail because the "JSON" is invalid, since it is actually an HTML. This can be solved here by adding a catch or finally.